packages/core/src/components/sections/ProductGallery/ProductGallery.tsx

```diff
@@ -24,28 +25,17 @@ export interface ProductGallerySectionProps {
   productCard?: ProductGalleryProps['productCard']
 }
 
-const isSearch = (x: any): x is SearchPageContextType =>
-  x === undefined || x?.title != undefined || x?.searchTerm != undefined
-const isCollection = (
-  x: any
-): x is ServerCollectionPageQueryQuery['collection'] =>
-  x?.seo != undefined && x?.seo != null && x?.sku == undefined
-
-function ProductGallerySection({
-  context,
-  ...otherProps
-}: ProductGallerySectionProps) {
-  const [title, searchTerm] = isSearch(context)
-    ? [context?.title, context?.searchTerm]
-    : isCollection(context)
-    ? [context?.seo?.title]
+function ProductGallerySection({ ...otherProps }: ProductGallerySectionProps) {
+  const context = usePage() as SearchPageContext | ProductListingPageContext
+  const [title, searchTerm] = isSearchPage(context)
+    ? [context?.data?.title, context?.data?.searchTerm]
+    : isPLP(context)
+    ? [context?.data?.collection?.seo?.title]
     : ['']
```

Comment:
Could we simplify to this?

```suggestion
 const searchTerm = context?.data?.searchTerm;
 const title = context?.data?.title ?? context?.data?.collection?.seo?.title ?? ''
```

Reply:
the same regarding the ts. we need to know the types otherwise the typescript complains.